### ISSUE TITLE
Passing parameter to ssh-plugin

### ISSUE THREAD

**Issue by markft** [cronicle-edge/cronicle-edge#75]
When using the workflow plugin to start a ssh-plugin the parameter is taken as the URI for the ssh connection.  I couldn't find a method to pass a parameter all the way to the script on the ssh destination.  My goal was to use JOB_ARG to pass a user defined parameter to the script on the destination host.  I made 2 modifications to ssh-plugin.js to allow workflows and the API call "run_event" to pass a parameter to the script: 1 removed JOB_ARG from the hostInfo variable; 2 added LC_JOB_ARG to the list of variable sent to the ssh session.

--- ssh-plugin.js.orig	2024-01-09 18:36:25.939424553 -0500
+++ ssh-plugin.js	2024-01-09 18:36:15.227242551 -0500
@@ -11,7 +11,7 @@
 	process.stdout.write(text + EOL);
 }
 
-let hostInfo = process.env['JOB_ARG'] || process.env['SSH_HOST'] || ''
+let hostInfo = process.env['SSH_HOST'] || ''
 
 let json = parseInt(process.env['JSON'] || '')
 
@@ -141,6 +141,7 @@
             LC_TMP_FILE: tmpFile
             , LC_JOB_ID: process.env['JOB_ID']
             , LC_BASE_URL: process.env['BASE_URL']
+            , LC_JOB_ARG: process.env['JOB_ARG'] || ''
         }

**Comment by mikeTWC1984** [comment]
I recall I was playing with LC variables, but the behavior wasn't consistent.  There is no guarantee I can push env variables to target, so I can't rely on it in plugin logic. I  used JOB_ARG as a reference to sftp by design, this way using WF plugin you can execute same code on multiple ssh targets.
 I'll try to set an option to generate LC variables

**Comment by ccxuy** [comment]
Is this support now?

**Comment by mikeTWC1984** [comment]
@ccxuy  yes, newer version of SSH plugin is pushing variables with prefix JOB_ and SSH_ to target host (those just attached on the bootstrap script)

**Comment by ccxuy** [comment]
Thank you, how could I use it? I want to assgin the same task to different machine by a single ssh job.

**Comment by ccxuy** [comment]
> [@ccxuy](https://github.com/ccxuy)* yes, newer version of SSH plugin is pushing variables with prefix JOB_ and SSH_ to target host (those just attached on the bootstrap script)

Seems like SSH_HOST occupy the whole JOB_ARG, and could not pass other arguments to workflow jobs? Is it possible to do something like this?

workflow:
- job1:
  arguments;
    ssh_host: a
    other_args:
- job2:
    ssh_host: b
    other_args:

**Comment by mikeTWC1984** [comment]
I think it works as you describe. If you specify target host via "SSH Host" parameter, the argument from workflow will be treated as argument ( it will be treated as host is that "SSH Host" is empty). I recall doing changes like this, but perhaps it was for Docker plugin, I'll check
